src/compiler/emitter.ts

```diff
@@ -5685,10 +5685,10 @@ var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, Promi
             function emitDecoratorsOfConstructor(node: ClassLikeDeclaration) {
                 const decorators = node.decorators;
                 const constructor = getFirstConstructorWithBody(node);
-                const hasDecoratedParameters = constructor && forEach(constructor.parameters, nodeIsDecorated);
+                const parameterDecorators = constructor && forEach(constructor.parameters, parameter => parameter.decorators);

```

Comment:
i see, never mind, i would rename this as firstParameterDecorator to avoid the confusion.